framework/src/meshlet_helpers.cpp

```diff
@@ -0,0 +1,139 @@
+﻿#include <gvk.hpp>
+
+namespace gvk
+{
+	std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<model_t>, std::vector<mesh_index_t>>>& aModels,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		return divide_into_meshlets(aModels, divide_into_very_bad_meshlets, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+	}
+
+	std::vector<meshlet> divide_into_meshlets_cached(gvk::serializer& aSerializer,
+		const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		return divide_into_meshlets_cached(aSerializer, aModels, divide_into_very_bad_meshlets, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+	}
+
+	template <typename F>
+	std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels, F&& aMeshletDivision,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		std::vector<meshlet> meshlets;
+		for (auto& pair : aModels) {
+			auto model = std::get<avk::resource_ownership<model_t>>(pair);
+			auto& meshIndices = std::get<std::vector<mesh_index_t>>(pair);
+
+			if (aCombineSubmeshes) {
+				auto [vertices, indices] = get_vertices_and_indices(std::vector({ std::make_tuple(avk::const_referenced(model.get()), meshIndices) }));
+				// default to very bad meshlet generation
+				std::vector<meshlet> tmpMeshlets = divide_vertices_into_meshlets(vertices, indices, model, std::nullopt, aMaxVertices, aMaxIndices, aMeshletDivision);
+				// append to meshlets
+				meshlets.insert(std::end(meshlets), std::begin(tmpMeshlets), std::end(tmpMeshlets));
+			}
+			else {
+				for (const auto meshIndex : meshIndices) {
+					auto vertices = model.get().positions_for_mesh(meshIndex);
+					auto indices = model.get().indices_for_mesh<uint32_t>(meshIndex);
+					// default to very bad meshlet generation
+					std::vector<meshlet> tmpMeshlets = divide_vertices_into_meshlets(vertices, indices, model, meshIndex, aMaxVertices, aMaxIndices, aMeshletDivision);
```

Comment:
```suggestion
					std::vector<meshlet> tmpMeshlets = divide_vertices_into_meshlets(vertices, indices, std::move(model), meshIndex, aMaxVertices, aMaxIndices, aMeshletDivision);
```

Reply:
Model assignment into `struct meshlet` happens in `divide_vertices_into_meshlets`, problem solved, I guess.

Reply:
same here